includes/admin/class-wc-stripe-settings-controller.php

```diff
@@ -62,11 +62,12 @@ public function admin_scripts() {
 		}
 
 		$params = [
-			'time'             => time(),
-			'i18n_out_of_sync' => wp_kses(
+			'time'              => time(),
+			'i18n_out_of_sync'  => wp_kses(
 				__( '<strong>Warning:</strong> your site\'s time does not match the time on your browser and may be incorrect. Some payment methods depend on webhook verification and verifying webhooks with a signing secret depends on your site\'s time being correct, so please check your site\'s time before setting a webhook secret. You may need to contact your site\'s hosting provider to correct the site\'s time.', 'woocommerce-gateway-stripe' ),
 				[ 'strong' => [] ]
 			),
+			'upe_setting_value' => isset( $gateway_settings['upe_checkout_experience_enabled'] ) ? 'no' : $gateway_settings['upe_checkout_experience_enabled'],
```

Comment:
`$gateway_settings` appears to be undefined here. I believe there is a filter named `wc_stripe_settings` if you want to hook into it or we can grab UPE settings from `WC_Stripe_Feature_Flags::is_upe_enabled()`

Which I guess brings up another point... Should the `WC_Stripe_Feature_Flags::is_upe_enabled()` method hook into the `wc_stripe_settings` filter? 🤔

Reply:
> `$gateway_settings` appears to be undefined here

Ah, the joys of merge conflicts :D thanks, fixed in dc1f7ce 👍 

> I believe there is a filter named `wc_stripe_settings` if you want to hook into it or we can grab UPE settings from `WC_Stripe_Feature_Flags::is_upe_enabled()`

It looks like the filter is used to grab the settings' configuration, not the settings' values 

> Should the `WC_Stripe_Feature_Flags::is_upe_enabled()` method hook into the `wc_stripe_settings` filter since it just grabs Stripe settings directly from the options table? 🤔 

`WC_Stripe_Feature_Flags::is_upe_enabled()` checks whether or not the user is eligible to _see_ the UPE configuration (i.e.: it controls the ability to enable/disable UPE, it doesn't enable/disable UPE checkout itself), whereas the `woocommerce_stripe_settings.upe_checkout_experience_enabled` controls whether the UPE checkout is enabled/disabled.

IMO `WC_Stripe_Feature_Flags::is_upe_enabled()` and `woocommerce_stripe_settings.upe_checkout_experience_enabled` should stay separate, since one controls the ability to control UPE and the other one controls UPE itself.

Maybe one thing we can do is add an _additional_ method on `WC_Stripe_Feature_Flags`?